### PR TITLE
updating fold function

### PR DESCRIPTION
**Pull request by Zainab-Abbas**
changed PartialAgg class in WindowGraphAggregation.java

### CODE REVIEW

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 1]
```diff
@@ -1,9 +1,7 @@
 package org.apache.flink.graph.streaming;
 
-import org.apache.flink.api.common.functions.FoldFunction;
-import org.apache.flink.api.common.functions.MapFunction;
-import org.apache.flink.api.common.functions.ReduceFunction;
-import org.apache.flink.api.common.functions.RichMapFunction;
+import org.apache.flink.api.common.functions.*;
+import org.apache.flink.api.common.functions.util.FunctionUtils;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: unused import

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
 import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
 import org.apache.flink.api.common.typeinfo.TypeInformation;
 import org.apache.flink.api.java.operators.translation.WrappingFunction;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: unused import

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 3]
```diff
@@ -1,9 +1,7 @@
 package org.apache.flink.graph.streaming;
 
-import org.apache.flink.api.common.functions.FoldFunction;
-import org.apache.flink.api.common.functions.MapFunction;
-import org.apache.flink.api.common.functions.ReduceFunction;
-import org.apache.flink.api.common.functions.RichMapFunction;
+import org.apache.flink.api.common.functions.*;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: can you please remove the wildcard import and import the required functions separately?

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 4]
```diff
@@ -82,22 +80,30 @@ public void open(Configuration parameters) throws Exception {
     }
 
     @SuppressWarnings("serial")
-    private static final class PartialAgg<K, EV, S> extends WrappingFunction<EdgesFold<K, EV, S>>
+    private static final class PartialAgg<K, EV, S>
             implements ResultTypeQueryable<S>, FoldFunction<Tuple2<Integer, Edge<K, EV>>, S> {
+        private static final long serialVersionUID = 1L;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: this is not needed

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 5]
```diff
@@ -82,22 +80,30 @@ public void open(Configuration parameters) throws Exception {
...
+        protected EdgesFold<K, EV, S> foldFunction;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: why protected and not private?

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 6]
```diff
@@ -82,22 +80,30 @@ public void open(Configuration parameters) throws Exception {
...
+
+
 
         public PartialAgg(EdgesFold<K, EV, S> foldFunction) {
-            super(foldFunction);
+            this.foldFunction=foldFunction;
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: fix format please

File: src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java [hunk 7]
```diff
@@ -82,22 +80,30 @@ public void open(Configuration parameters) throws Exception {
...
         }
 
         @Override
         public TypeInformation<S> getProducedType() {
-            return TypeExtractor.createTypeInfo(EdgesFold.class, getWrappedFunction().getClass(), 2,
+            return TypeExtractor.createTypeInfo(EdgesFold.class, getFunction().getClass(), 2,
                     null, null);
         }
+
+        public EdgesFold<K, EV, S> getFunction() {
```
Comment (vasia) [inline src/main/java/org/apache/flink/graph/streaming/WindowGraphAggregation.java]: This method is not needed. You can directly use `foldFunction`.

**Comment by vasia** [comment]
Thank you @Zainab-Abbas! I have left a few easy-to-fix comments.

**Comment by Zainab-Abbas** [comment]
made some small fixes :)

**Comment by vasia** [comment]
Thanks @Zainab-Abbas! Changes look good. Can you please revert all unrelated formatting changes? It seems that your IDE likes adding tabs :)
I'll merge afterwards.

**Comment by Zainab-Abbas** [comment]
@vasia I do not know why IDE keeps adding spaces every time i format it,
its adding even more spaces
my current settings:

use tab character option checked

Tab size: 4
Indent: 4
Continuation indent: 8

let me know if this has to be changed

**Comment by Zainab-Abbas** [comment]
ok i got the issue the pervious WindowGraphAggregation.java is not formatted correctly
this one is correct as the IDE gives notification about previous one not being indented properly.

I can revert it back to fewer spaces if you want, but i checked other code files they are according to my current commit.

**Pull request closed without merging**